packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,58 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
```

Comment:
Should this be moved to DateUtils so we can unit test this?

Reply:
I'm updating things to remove this. The overhaul should take care of quite a bit of stuff like this!